src/community/backup-restore/core/src/main/java/org/geoserver/backuprestore/tasklet/AbstractCatalogBackupRestoreTasklet.java

```diff
@@ -0,0 +1,342 @@
+import org.springframework.batch.core.StepContribution;
+import org.springframework.batch.core.scope.context.ChunkContext;
+import org.springframework.batch.core.step.tasklet.Tasklet;
+import org.springframework.batch.repeat.RepeatStatus;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.util.Assert;
+
+/**
+ * TODO
+ * 
+ * @author Alessio Fabiani, GeoSolutions
+ *
+ */
+@SuppressWarnings("rawtypes")
+public abstract class AbstractCatalogBackupRestoreTasklet<T> extends BackupRestoreItem
+        implements Tasklet, InitializingBean {
+
+    protected static Logger LOGGER = Logging.getLogger(AbstractCatalogBackupRestoreTasklet.class);
+
+    /*
+     * 
+     */
+    protected static Map<String, Filter<Resource>> resources = new HashMap<String, Filter<Resource>>();
+
+    /*
+     * 
+     */
+    static {
+        resources.put("demo", AnyFilter.INSTANCE);
+        resources.put("images", AnyFilter.INSTANCE);
+        resources.put("logs", new Filter<Resource>() {
+
+            @Override
+            public boolean accept(Resource res) {
+                if (res.name().endsWith(".properties")) {
+                    return true;
+                }
+                return false;
+            }
+
+        });
+        resources.put("palettes", AnyFilter.INSTANCE);
+        resources.put("plugIns", AnyFilter.INSTANCE);
+        resources.put("styles", new Filter<Resource>() {
+
+            @Override
+            public boolean accept(Resource res) {
+                if (res.name().endsWith(".sld") || res.name().endsWith(".xml")) {
+                    return false;
+                }
+                return true;
+            }
+
+        });
+        resources.put("user_projections", AnyFilter.INSTANCE);
+        resources.put("validation", AnyFilter.INSTANCE);
+        resources.put("www", AnyFilter.INSTANCE);
+    }
+
+    public AbstractCatalogBackupRestoreTasklet(Backup backupFacade,
+            XStreamPersisterFactory xStreamPersisterFactory) {
+        super(backupFacade, xStreamPersisterFactory);
+    }
+
+    @Override
+    public RepeatStatus execute(StepContribution contribution, ChunkContext chunkContext)
+            throws Exception {
+        super.retrieveInterstepData(chunkContext.getStepContext().getStepExecution());
+        JobExecution jobExecution = chunkContext.getStepContext().getStepExecution()
+                .getJobExecution();
+
+        return doExecute(contribution, chunkContext, jobExecution);
+    }
+
+    /**
+     * 
+     * @param contribution
+     * @param chunkContext
+     * @param jobExecution
+     * @return
+     * @throws Exception
+     */
+    abstract RepeatStatus doExecute(StepContribution contribution, ChunkContext chunkContext,
+            JobExecution jobExecution) throws Exception;
+
+    /**
+     * @param targetBackupFolder
+     * @throws Exception
+     */
+    public void backupGWCSettings(Resource targetBackupFolder) throws Exception {
```

Comment:
Nope, it is used by all implementations of Tasklets now.

See also  https://github.com/geosolutions-it/geoserver/blob/backup_restore/src/community/backup-restore/core/src/main/java/org/geoserver/backuprestore/tasklet/CatalogSecurityManagerTasklet.java